src/BaseFirestoreRepository.ts

```diff
@@ -50,12 +50,18 @@ export class BaseFirestoreRepository<T extends IEntity>
   }
 
   async create(item: T): Promise<T> {
+    if (this.config.validate) {
```

Comment:
Open question: The check to run the validation shouldn't be inside `validate` so we only have to do it once? That makes the function impure (adds side effects) but that is a private function anyway.

Reply:
I thought that at first, but I thought it would be easier to understand if it was written like this. I'm happy either way though

Reply:
I prefer the dirty / with side effects one

Reply:
Okay, maybe change the function to `maybeValidate` then?

Reply:
That's even more confusing 😂. Let's keep it as is. I don't foresee any other method calling this function other than `create` and `update`